test/Crystallization.cpp

```diff
@@ -0,0 +1,177 @@
+
+	cadet::test::column::Dummyparams disc; // CSTR, so no spatial resolution
+	cadet::test::column::testReferenceBenchmark(modelFilePath, refFilePath, "001", absTol, relTol, disc, true);
+}
+
+TEST_CASE("Crystallization in a CSTR with primary nucleation and growth", "[Crystallization],[Simulation],[Reference],[CI]")
+{
+	const std::string& modelFilePath = std::string("/data/configuration_PBM_CSTR_primaryNucleationAndGrowth_benchmark1.json");
+	const std::string& refFilePath = std::string("/data/ref_PBM_CSTR_primaryNucleationAndGrowth_benchmark1.h5");
+	const std::vector<double> absTol = { 1e-10 };
+	const std::vector<double> relTol = { 1e-10 };
+
+	cadet::test::column::Dummyparams disc; // CSTR, so no spatial resolution
+	cadet::test::column::testReferenceBenchmark(modelFilePath, refFilePath, "001", absTol, relTol, disc, true);
+}
+
+TEST_CASE("Crystallization in a CSTR with primary nucleation, growth and growth rate dispersion", "[Crystallization],[Simulation],[Reference],[CI]")
+{
+	const std::string& modelFilePath = std::string("/data/configuration_PBM_CSTR_primaryNucleationGrowthGrowthRateDispersion_benchmark1.json");
+	const std::string& refFilePath = std::string("/data/ref_PBM_CSTR_primaryNucleationGrowthGrowthRateDispersion_benchmark1.h5");
+	const std::vector<double> absTol = { 1e-10 };
+	const std::vector<double> relTol = { 1e-10 };
+
+	cadet::test::column::Dummyparams disc; // CSTR, so no spatial resolution
+	cadet::test::column::testReferenceBenchmark(modelFilePath, refFilePath, "001", absTol, relTol, disc, true);
+}
+
+TEST_CASE("Crystallization in a CSTR with primary and secondary nucleation and growth", "[Crystallization],[Simulation],[Reference],[CI]")
+{
+	const std::string& modelFilePath = std::string("/data/configuration_PBM_CSTR_primarySecondaryNucleationAndGrowth_benchmark1.json");
+	const std::string& refFilePath = std::string("/data/ref_PBM_CSTR_primarySecondaryNucleationAndGrowth_benchmark1.h5");
+	const std::vector<double> absTol = { 1e-9 };
+	const std::vector<double> relTol = { 1e-6 };
+
+	cadet::test::column::Dummyparams disc; // CSTR, so no spatial resolution
+	cadet::test::column::testReferenceBenchmark(modelFilePath, refFilePath, "001", absTol, relTol, disc, true);
+}
+
+TEST_CASE("Crystallization in a DPFR/LRM with primary and secondary nucleation and growth", "[Crystallization],[Simulation],[Reference],[CI]")
+{
+	const std::string& modelFilePath = std::string("/data/configuration_PBM_DPFR_primarySecondaryNucleationGrowth_benchmark1.json");
+	const std::string& refFilePath = std::string("/data/ref_PBM_DPFR_primarySecondaryNucleationGrowth_benchmark1.h5");
+	const std::vector<double> absTol = { 7e+6 };
+	const std::vector<double> relTol = { 1e-10 };
+
+	cadet::test::column::FVparams disc(25);
+	cadet::test::column::testReferenceBenchmark(modelFilePath, refFilePath, "001", absTol, relTol, disc, true);
+}
+
+// Note: two entries fail, given in (row,col): (204, 190), (204, 191)
```

Comment:
@WFlynnZ some Jacobian tests fail for only two entries, like the one here. Seems to be the same two entries for four tests cases